pkg/handlers/embedded_cluster_artifacts.go

```diff
@@ -114,3 +113,104 @@ func (h *Handler) GetEmbeddedClusterBinary(w http.ResponseWriter, r *http.Reques
 		return
 	}
 }
+
+// GetEmbeddedClusterInfraImages returns the infrastructure images as a .tgz file
+func (h *Handler) GetEmbeddedClusterInfraImages(w http.ResponseWriter, r *http.Request) {
+	if !util.IsEmbeddedCluster() {
+		logger.Error(errors.New("not an embedded cluster"))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Get k0s directory path from env var
+	k0sDir := util.EmbeddedClusterK0sDir()
+	if k0sDir == "" {
+		logger.Error(errors.New("environment variable EMBEDDED_CLUSTER_K0S_DIR not set"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Path to images directory
+	imagesDir := filepath.Join(k0sDir, "images")
+
+	// Check if images directory exists
+	_, err := os.Stat(imagesDir)
+	if os.IsNotExist(err) {
+		logger.Error(errors.Wrap(err, "images directory not found"))
+		w.WriteHeader(http.StatusNotFound)
+		return
+	} else if err != nil {
+		logger.Error(errors.Wrap(err, "failed to stat images directory"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Find all infra image files
+	entries, err := os.ReadDir(imagesDir)
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to read images directory"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Filter and sort image files
+	var infraImages []os.FileInfo
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		// Check if it's an infra images file
+		if !strings.HasSuffix(entry.Name(), "images-amd64.tar") {
+			continue
+		}
+
+		// Get file info for sorting by modification time
+		fileInfo, err := entry.Info()
+		if err != nil {
+			logger.Error(errors.Wrapf(err, "failed to get file info for %s", entry.Name()))
+			continue
+		}
+
+		fmt.Println("fileInfo", fileInfo.Name(), fileInfo.ModTime())
+
+		infraImages = append(infraImages, fileInfo)
+	}
+
+	if len(infraImages) == 0 {
+		logger.Error(errors.New("no infrastructure image files found"))
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	// Sort by modification time, newest first
+	sort.Slice(infraImages, func(i, j int) bool {
+		return infraImages[i].ModTime().After(infraImages[j].ModTime())
```

Comment:
Can you give me an example of the file names you expect?

Reply:
`images-amd64.tar` on install, and `ec-images-amd64.tar` on upgrade. we should consolidate those names.

Reply:
Perhaps a comment explaining that?